README.md
# Scala Exercises - Fetch library

This repository hosts a content library for the [Scala Exercises](https://www.scala-exercises.org/) platform, including interactive exercises related to the [Fetch](https://github.com/47deg/fetch) data access library.

## About Scala exercises

"Scala Exercises" brings exercises for the Stdlib, Cats, Shapeless and many other great libraries for Scala to your browser. Offering hundreds of solvable exercises organized into several categories covering the basics of the Scala language and it's most important libraries.

Scala Exercises is available at [scala-exercises.org](https://scala-exercises.org).

## Contributing

Contributions welcome! Please join our [Gitter channel](https://gitter.im/scala-exercises/scala-exercises)
to get involved, or visit our [GitHub site](https://github.com/scala-exercises).

## License

Copyright (C) 2015-2016 47 Degrees, LLC.
Reactive, scalable software solutions.
http://47deg.com
hello@47deg.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.


CHANGELOG.md
# Changelog

## [v0.6.6](https://github.com/scala-exercises/exercises-fetch/tree/v0.6.6) (2020-08-10)

[Full Changelog](https://github.com/scala-exercises/exercises-fetch/compare/v0.6.5...v0.6.6)

📈 **Dependency updates**

- Update definitions, exercise-compiler, ... to 0.6.4 [\#71](https://github.com/scala-exercises/exercises-fetch/pull/71) ([47erbot](https://github.com/47erbot))

## [v0.6.5](https://github.com/scala-exercises/exercises-fetch/tree/v0.6.5) (2020-08-10)

[Full Changelog](https://github.com/scala-exercises/exercises-fetch/compare/v0.6.4...v0.6.5)

📈 **Dependency updates**

- Update sbt-mdoc to 2.2.4 [\#70](https://github.com/scala-exercises/exercises-fetch/pull/70) ([47erbot](https://github.com/47erbot))
- Update scalacheck-1-14 to 3.2.1.0 [\#69](https://github.com/scala-exercises/exercises-fetch/pull/69) ([47erbot](https://github.com/47erbot))
- Update scalatest to 3.2.1 [\#68](https://github.com/scala-exercises/exercises-fetch/pull/68) ([47erbot](https://github.com/47erbot))
- Update sbt-scalafmt to 2.4.2 [\#67](https://github.com/scala-exercises/exercises-fetch/pull/67) ([47erbot](https://github.com/47erbot))
- Update scalafmt-core to 2.6.4 [\#66](https://github.com/scala-exercises/exercises-fetch/pull/66) ([47erbot](https://github.com/47erbot))
- Update scalafmt-core to 2.6.3 [\#65](https://github.com/scala-exercises/exercises-fetch/pull/65) ([scala-steward](https://github.com/scala-steward))
- Update scalafmt-core to 2.6.2 [\#64](https://github.com/scala-exercises/exercises-fetch/pull/64) ([scala-steward](https://github.com/scala-steward))

## [v0.6.4](https://github.com/scala-exercises/exercises-fetch/tree/v0.6.4) (2020-06-30)

[Full Changelog](https://github.com/scala-exercises/exercises-fetch/compare/v0.6.3...v0.6.4)

## [v0.6.3](https://github.com/scala-exercises/exercises-fetch/tree/v0.6.3) (2020-06-29)

[Full Changelog](https://github.com/scala-exercises/exercises-fetch/compare/v0.6.2...v0.6.3)

📈 **Dependency updates**

- Update sbt to 1.3.13 [\#58](https://github.com/scala-exercises/exercises-fetch/pull/58) ([scala-steward](https://github.com/scala-steward))
- Update sbt-mdoc to 2.2.3 [\#57](https://github.com/scala-exercises/exercises-fetch/pull/57) ([scala-steward](https://github.com/scala-steward))
- Update scalafmt-core to 2.6.1 [\#56](https://github.com/scala-exercises/exercises-fetch/pull/56) ([scala-steward](https://github.com/scala-steward))
- Update scalacheck-1-14 to 3.2.0.0 [\#55](https://github.com/scala-exercises/exercises-fetch/pull/55) ([scala-steward](https://github.com/scala-steward))
- Update scalatest to 3.2.0 [\#54](https://github.com/scala-exercises/exercises-fetch/pull/54) ([scala-steward](https://github.com/scala-steward))
- Update sbt-mdoc to 2.2.2 [\#47](https://github.com/scala-exercises/exercises-fetch/pull/47) ([scala-steward](https://github.com/scala-steward))
- Update sbt to 1.3.11 [\#45](https://github.com/scala-exercises/exercises-fetch/pull/45) ([scala-steward](https://github.com/scala-steward))
- Update scalatest to 3.1.2 [\#44](https://github.com/scala-exercises/exercises-fetch/pull/44) ([scala-steward](https://github.com/scala-steward))

**Merged pull requests:**

- Updates build [\#60](https://github.com/scala-exercises/exercises-fetch/pull/60) ([juanpedromoreno](https://github.com/juanpedromoreno))
- Update scalafmt-core to 2.6.0 [\#52](https://github.com/scala-exercises/exercises-fetch/pull/52) ([BenFradet](https://github.com/BenFradet))
- Update scalafmt-core to 2.5.3 [\#49](https://github.com/scala-exercises/exercises-fetch/pull/49) ([BenFradet](https://github.com/BenFradet))
- Prepare repository for next  release and SBT build improvements [\#46](https://github.com/scala-exercises/exercises-fetch/pull/46) ([juanpedromoreno](https://github.com/juanpedromoreno))

## [v0.6.2](https://github.com/scala-exercises/exercises-fetch/tree/v0.6.2) (2020-04-27)

[Full Changelog](https://github.com/scala-exercises/exercises-fetch/compare/v0.6.1.1...v0.6.2)

## [v0.6.1.1](https://github.com/scala-exercises/exercises-fetch/tree/v0.6.1.1) (2020-04-27)

[Full Changelog](https://github.com/scala-exercises/exercises-fetch/compare/v0.6.1...v0.6.1.1)

## [v0.6.1](https://github.com/scala-exercises/exercises-fetch/tree/v0.6.1) (2020-04-27)

[Full Changelog](https://github.com/scala-exercises/exercises-fetch/compare/v0.6.0...v0.6.1)

**Merged pull requests:**

- Compile when Publish [\#39](https://github.com/scala-exercises/exercises-fetch/pull/39) ([juanpedromoreno](https://github.com/juanpedromoreno))

## [v0.6.0](https://github.com/scala-exercises/exercises-fetch/tree/v0.6.0) (2020-04-24)

[Full Changelog](https://github.com/scala-exercises/exercises-fetch/compare/34fe35f22303148eed59d2a15817405d00032518...v0.6.0)

**Closed issues:**

- New Fetch logo [\#14](https://github.com/scala-exercises/exercises-fetch/issues/14)

**Merged pull requests:**

- Updates fetch [\#36](https://github.com/scala-exercises/exercises-fetch/pull/36) ([juanpedromoreno](https://github.com/juanpedromoreno))
- Updates Project [\#31](https://github.com/scala-exercises/exercises-fetch/pull/31) ([juanpedromoreno](https://github.com/juanpedromoreno))
- Update sbt to 1.3.10 [\#29](https://github.com/scala-exercises/exercises-fetch/pull/29) ([scala-steward](https://github.com/scala-steward))
- Update sbt to 1.3.9 [\#28](https://github.com/scala-exercises/exercises-fetch/pull/28) ([scala-steward](https://github.com/scala-steward))
- Update fetch, fetch-debug to 1.2.2 [\#27](https://github.com/scala-exercises/exercises-fetch/pull/27) ([scala-steward](https://github.com/scala-steward))
- Update scalacheck-shapeless\_1.14 to 1.2.5 [\#26](https://github.com/scala-exercises/exercises-fetch/pull/26) ([scala-steward](https://github.com/scala-steward))
- Update scalatest to 3.1.1 [\#25](https://github.com/scala-exercises/exercises-fetch/pull/25) ([scala-steward](https://github.com/scala-steward))

project/ProjectPlugin.scala
import sbt.Keys._
import sbt._
import com.alejandrohdezma.sbt.github.SbtGithubPlugin

object ProjectPlugin extends AutoPlugin {

  override def trigger: PluginTrigger = allRequirements

  override def requires: Plugins = plugins.JvmPlugin && SbtGithubPlugin

  object autoImport {

    lazy val V = new {
      val cats: String                = "2.1.1"
      val fetch: String               = "1.3.0"
      val shapeless: String           = "2.3.3"
      val scala: String               = "2.13.2"
      val scalaExercises: String      = "0.6.0"
      val scalacheck: String          = "1.14.3"
      val scalacheckShapeless: String = "1.2.5"
      val scalatest: String           = "3.1.2"
      val scalatestplusScheck: String = "3.1.1.1"
    }

    def dep(artifactId: String) = "org.scala-exercises" %% artifactId % V.scalaExercises

    lazy val exercisesSettings = Seq(
      libraryDependencies ++= Seq(
        dep("exercise-compiler"),
        dep("definitions"),
        "org.typelevel"              %% "cats-core"                 % V.cats,
        "com.47deg"                  %% "fetch"                     % V.fetch,
        "com.47deg"                  %% "fetch-debug"               % V.fetch,
        "com.chuusai"                %% "shapeless"                 % V.shapeless,
        "org.scalatest"              %% "scalatest"                 % V.scalatest,
        "org.scalacheck"             %% "scalacheck"                % V.scalacheck,
        "org.scalatestplus"          %% "scalacheck-1-14"           % V.scalatestplusScheck,
        "com.github.alexarchambault" %% "scalacheck-shapeless_1.14" % V.scalacheckShapeless
      )
    )
  }

  override def projectSettings: Seq[Def.Setting[_]] =
    Seq(
      organization := "org.scala-exercises",
      organizationName := "47 Degrees",
      organizationHomepage := Some(url("https://47deg.com")),
      scalaVersion := autoImport.V.scala,
      resolvers ++= Seq(
        Resolver.sonatypeRepo("snapshots"),
        Resolver.sonatypeRepo("releases")
      )
    )
}
